ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/Overview/ClustersTable.tsx

```diff
@@ -118,29 +113,31 @@ function ClustersTable({ querySearchFilter, isFiltered, pagination }: ClustersTa
                     <Th>Kubernetes version</Th>
                 </Tr>
             </Thead>
-            {tableState.type === 'LOADING' && <TbodyLoading colSpan={colSpan} />}
-            {tableState.type === 'ERROR' && (
-                <TbodyError colSpan={colSpan} error={tableState.error} />
-            )}
-            {tableState.type === 'EMPTY' && (
-                <TbodyEmpty colSpan={colSpan} message="No secured clusters have been detected" />
-            )}
-            {tableState.type === 'FILTERED_EMPTY' && <TbodyFilteredEmpty colSpan={colSpan} />}
-            {tableState.type === 'COMPLETE' &&
-                tableState.data.map(({ id, name, clusterVulnerabilityCount, type, status }) => (
-                    <Tbody key={id}>
-                        <Tr>
-                            <Td dataLabel="Cluster" modifier="nowrap">
-                                <Link to={getPlatformEntityPagePath('Cluster', id)}>{name}</Link>
-                            </Td>
-                            <Td dataLabel="CVEs">{pluralize(clusterVulnerabilityCount, 'CVE')}</Td>
-                            <Td dataLabel="Cluster type">{displayClusterType(type)}</Td>
-                            <Td dataLabel="Kubernetes version">
-                                {status?.orchestratorMetadata?.version ?? 'UNKNOWN'}
-                            </Td>
-                        </Tr>
-                    </Tbody>
-                ))}
+            <TbodyTableState
+                tableState={tableState}
+                colSpan={colSpan}
+                emptyProps={{ message: 'No secured clusters have been detected' }}
+                renderWith={({ data }) =>
+                    data.map(({ id, name, clusterVulnerabilityCount, type, status }) => (
+                        <Tbody key={id}>
+                            <Tr>
+                                <Td dataLabel="Cluster" modifier="nowrap">
+                                    <Link to={getPlatformEntityPagePath('Cluster', id)}>
+                                        {name}
+                                    </Link>
+                                </Td>
+                                <Td dataLabel="CVEs">
+                                    {pluralize(clusterVulnerabilityCount, 'CVE')}
+                                </Td>
+                                <Td dataLabel="Cluster type">{displayClusterType(type)}</Td>
+                                <Td dataLabel="Kubernetes version">
+                                    {status?.orchestratorMetadata?.version ?? 'UNKNOWN'}
```

Comment:
Off the subject, **Unavailable** seems like existing pattern for this situation.

At the risk of seeming negative, for the candidate use case to include something as unusual as user-facing **UNKNOWN** in ALL CAPS raises the question how long boilerplate-saving patterns can survive continuous creep to complications?

Reply:
Ouch - yes "Unavailable" is a much better candidate here instead of "UNKNOWN". I'll make this change regardless of what happens with the pattern in this PR.

I'm hoping that the pattern here will be somewhat resilient to these types of complications. The core of this use case is that it abstracts away the checking of "LOADING", "ERROR", and rendering of the appropriate component which would otherwise be mostly copy-paste code. At least as it pertains to VM 2.0, we will have roughly 10-15 places where that pattern has or will emerge. If there are exceptions where we need to stray from the plan in the design doc as it pertains to rendering table states, this component is composed of the individual designs from the doc that we could use directly instead.

My hope is that the render prop pattern here can avoid complications by making them external to the template, and the caller's responsibility when they occur. e.g. Cases like this where we need to handle missing values, or cases where we want to conditionally render an action column, a checkbox, RBAC-dependent-data, etc, would be the responsibility of whatever parent component is using the template. I also hope that this pattern will help us avoid the awkwardness of checking if `data` is `undefined` when rendering the happy path, as in order to get to that path the `getTableUIState()` function and this component will have ensured that is not possible.

I don't have a concrete example in code, but I believe this would be resilient to the discussed `POLLING` and `LOADING_MORE` states as well. `POLLING` being something that IMO would use the same success state here while showing some piece of UI external to the table. `LOADING_MORE` would likely extend this pattern to pass an `isLoading`/`isFetching` prop along with `data` in the render props in order to implement whatever the intermediate state is. (One possibility is the `LiveTr`/`LiveTd` component pattern from [this WIP](https://github.com/stackrox/stackrox/pull/9134), but it still needs more discussion.)